node/activation_service.go

```diff
@@ -0,0 +1,712 @@
+	select {
+	case <-ctx.Done():
+		return nil
+	case err = <-app.errCh:
+		return err
+	}
+}
+
+func (app *App) startActivationServiceSynchronous(ctx context.Context) (err error) {
+	// notify anyone who might be listening that the app has finished starting.
+	// this can be used by, e.g., app tests.
+	defer close(app.started)
+
+	// Create a contextual logger for local usage (lower-level modules will create their own contextual loggers
+	// using context passed down to them)
+	logger := app.log.WithContext(ctx)
+
+	hostname, err := os.Hostname()
+	if err != nil {
+		return fmt.Errorf("error reading hostname: %w", err)
+	}
+
+	logger.With().Info("starting spacemesh",
+		log.String("data-dir", app.Config.DataDir()),
+		log.String("post-dir", app.Config.SMESHING.Opts.DataDir),
+		log.String("hostname", hostname),
+	)
+
+	if err := os.MkdirAll(app.Config.DataDir(), 0o700); err != nil {
+		return fmt.Errorf(
+			"data-dir %s not found or could not be created: %w",
+			app.Config.DataDir(),
+			err,
+		)
+	}
+
+	/* Setup monitoring */
+	app.errCh = make(chan error, 100)
+	if app.Config.PprofHTTPServer {
+		logger.With().Info("starting pprof server", log.String("address", app.Config.PprofHTTPServerListener))
+		app.pprofService = &http.Server{Addr: app.Config.PprofHTTPServerListener}
+		app.eg.Go(func() error {
+			if err := app.pprofService.ListenAndServe(); err != nil {
+				app.errCh <- fmt.Errorf("cannot start pprof http server: %w", err)
+			}
+			return nil
+		})
+		if app.Config.PprofMutexProfile {
+			// this will set the mutex profiling to sample a third of all lock events
+			runtime.SetMutexProfileFraction(3)
+		}
+		if app.Config.PprofBlockProfile {
+			// record block sample for every block event that takes more than 10 milliseconds
+			runtime.SetBlockProfileRate(int(10 * time.Millisecond))
+		}
+	}
+
+	if app.Config.ProfilerURL != "" {
+		app.profilerService, err = pyroscope.Start(pyroscope.Config{
+			ApplicationName: app.Config.ProfilerName,
+			// app.Config.ProfilerURL should be the pyroscope server address
+			// TODO: AuthToken? no need right now since server isn't public
+			ServerAddress: app.Config.ProfilerURL,
+			// by default all profilers are enabled,
+		})
+		if err != nil {
+			return fmt.Errorf("cannot start profiling client: %w", err)
+		}
+	}
+
+	var preserved *checkpoint.PreservedData
+	if app.Config.Recovery.Uri != "" {
+		preserved, err = app.loadCheckpoint(ctx)
+		if err != nil {
+			return fmt.Errorf("loading checkpoint: %w", err)
+		}
+	}
+
+	/* Initialize all protocol services */
+	app.clock, err = timesync.NewClock(
+		timesync.WithLayerDuration(app.Config.LayerDuration),
+		timesync.WithTickInterval(1*time.Second),
+		timesync.WithGenesisTime(app.Config.Genesis.GenesisTime.Time()),
+		timesync.WithLogger(app.addLogger(ClockLogger, logger).Zap()),
+	)
+	if err != nil {
+		return fmt.Errorf("cannot create clock: %w", err)
+	}
+
+	if err := app.setupDBs(ctx, logger); err != nil {
```

Comment:
For a followup probably: the `setupDBs()` sets up both state and local DBs, but only the local DB is needed on the smashing service.

Reply:
It would be nice to have that in node-split, but it seems to be not crucial from the start, so probably we should move that as a followup.
In either case IMO it requires tracking, so it should be added to https://github.com/spacemeshos/pm/issues/338 or to new issue which will cover followups.
@pigmej am I right?

Reply:
It IS crucial and essential to NOT open the unnecessary DB on the smeshing service. That's, I' 'd say, rather essential.
For me that was obvious that we need to get rid of the double DB opening on the smeshing-service "post PoC".

Reply:
should i start working on that in this PR or in a followup?

Reply:
It can be a follow up

Reply:
https://github.com/spacemeshos/go-spacemesh/issues/6639